### PR TITLE
(PXP-9835): feat/studyViewerIdpFlag: add flag to check for idp login type before allowing user to request access

### PR DESCRIPTION
**Pull request by ocshawn**
Jira Ticket: [PXP-9835](https://ctds-planx.atlassian.net/browse/PXP-9835)

### New Features
- Study viewer: add flag to check for idp login type before allowing user to request access

### CODE REVIEW

File: docs/study_viewer.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ Example configuration:
             {
                 // configuration specific to the "Request Access" button
                 "type": "request_access",
+                "required_idp": "ImmPort",// optional, requires user to have specific type to request access
```
Comment (paulineribeyre) [inline docs/study_viewer.md:76]:
```suggestion
                "required_idp_field": "required_idp", // optional, requires user to have logged in through this identity provider to request access
```

Comment (paulineribeyre) [reply docs/study_viewer.md:76]: The feature is a bit more complex than the current implementation: the config should be `required_idp_field`, which is set to the field in the dictionary that contains the required IDP - for NCT it's `required_idp`.

Each row of data could have `required_idp` set or not, for example one row could have it set to `immport`, another to `google` and another to nothing (so any IDP would work to request access to this one)

**Review by paulineribeyre — CHANGES_REQUESTED** [review]

**Comment by ocshawn** [comment]
changes have been made and code is in QA https://qa-niaid.planx-pla.net/ please let me know if it works as i cant log in with immport

**Comment by paulineribeyre** [comment]
@ocshawn it works in qa-niaid 🎉 fyi you can create an account in the immport dev system at https://user-admin.dev.immport.org/registrationuser/registration

File: src/StudyViewer/StudyDetails.jsx [hunk 2]
```diff
@@ -264,20 +272,24 @@ class StudyDetails extends React.Component {
...
-                   onClick={() => this.props.history.push(`${this.props.location.pathname}/${encodeURIComponent(this.props.data.rowAccessorValue)}`)}
-                 />
-               )
-               : null}
-             {
-               (this.props.studyViewerConfig.buttons) ? this.props.studyViewerConfig.buttons.map(
-                 (buttonConfig, i) => this.getButton(i, buttonConfig, userHasLoggedIn),
-               ) : null
-             }
+             <div>
```
Comment (paulineribeyre) [inline src/StudyViewer/StudyDetails.jsx:275]: is this change only a linting change?

File: src/StudyViewer/reduxer.js [hunk 3]
```diff
@@ -41,6 +41,10 @@ export const fetchFiles = (dataType, typeOfFileIndex, rowAccessorValue) => {
   }
   let nameOfIndex;
   const fieldsToFetch = ['file_name', 'file_size', 'data_format', 'data_type', targetStudyViewerConfig.rowAccessor];
+  const requiredIdpField = targetStudyViewerConfig.buttons.find((obj) => obj.type === 'request_access')?.required_idp_field;
+  if (requiredIdpField) {
+    fieldsToFetch.push(requiredIdpField);
+  }
```
Comment (paulineribeyre) [inline src/StudyViewer/reduxer.js:47]: why do we need to add this to `fetchFiles`? i don't think documents in those indices will have the `required_idp` field

File: src/StudyViewer/StudyDetails.jsx (hunk 2 above)
Comment (ocshawn) [reply src/StudyViewer/StudyDetails.jsx:275]: no i added it to eliminate a react duplicate key warning that was flooding the console, i tested the change and it had no visual impact of the page

File: src/StudyViewer/reduxer.js (hunk 3 above)
Comment (ocshawn) [reply src/StudyViewer/reduxer.js]: we dont, i removed it, dont know why it was added. Thanks for catching it

**Comment by ocshawn** [comment]
> @ocshawn it works in qa-niaid tada fyi you can create an account in the immport dev system at https://user-admin.dev.immport.org/registrationuser/registration

thanks i signed up previously but it took a day or two for it to work

**Review by paulineribeyre — DISMISSED** [review]

File: src/StudyViewer/StudyDetails.jsx (hunk 2 above)
Comment (ocshawn) [reply src/StudyViewer/StudyDetails.jsx]: had to remove the divs as it was failing automated testing, replaced with react fragments and hope that works

**Review by paulineribeyre — APPROVED** [review]

**Pull request merged**